mantidproject/mantid
# Adding feature flags into the muon GUIs

**Description of work.**
This work will make it easier for us to develop new muon features, while keeping them hidden from typical users. It will also allow us to give instructions to super users on how to activate new features for testing.

The unit tests will be done as part of https://github.com/mantidproject/mantid/issues/32632 and will be done later as this work is needed for other developers to continue their work



**To test:**


MA = muon analysis
FDA = frequency domain analysis

Open MA/FDA
The last tab will be results
The plotting options will be data and fit.

Close Mantid
Go into the user properties file and add the line:
`muon.GUI=model_analysis:2, raw_plots:0`
Open Mantid
Open MA
The last tab will be "model"
Do the superconducting test https://developer.mantidproject.org/Testing/MuonInterface/MuonTesting.html 
On the last tab you will be able to generate a plot that looks like the one in the doc
Open FDA
Load MUSR62260
Go to transform
Do an FFT
Do a couple of fits
Produce a results table (include the "group" from the sample logs)
On the model tab you should be able to make a plot of the group versus something else

Close Mantid
Go into the user properties file and edit the line:
`muon.GUI=model_analysis:1, raw_plots:0`
Open Mantid
Open MA
The last tab will be "model"
Do the superconducting test https://developer.mantidproject.org/Testing/MuonInterface/MuonTesting.html 
On the last tab you will be able to do a fit but it will not plot anything. You can inspect the fit from the ADS
Open FDA
Load MUSR62260
Go to transform
Do an FFT
Do a couple of fits
Produce a results table (include the "group" from the sample logs)
On the model tab you should be able to do a fit

Close Mantid
Go into the user properties file and edit the line:
Open Mantid
`muon.GUI=model_analysis:1, raw_plots:1`
Open MA/FFT
The last tab will be "model"
Load some data
In the plotting you can change the option to "raw"
It will show a bunch of tiled plots of the count data

Close Mantid 
Go into the user properties file and edit the line:
such that `muon.GUI=`  has an invalid input (mangle the string)
Open Mantid
It should all work ok, but you will not have model or raw plots

Fixes #32572. 



This is not to be advertised to users.


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
